### ISSUE TITLE
pytest and naming convention of modules and scripts

### ISSUE THREAD

**Issue by erwee6** [pmayd/open-discourse#10]
When creating a test_script I have to import the script that should be tested.

The imports are producing errors due to the leading digits of the module and script-file-names, 
e.g. 02_preprocessing and 01_split_xml. 
When I rename the module to M02_preprocessing and the script-file to P01_split_xml everything is fine. 
AFAIK module names have to be valid identifiers in python and identifiers must not start with a digit. 

Should we rename the modules and scripts (what prefix?) or am I missing something?

**Comment by tifa365** [comment]
I haven't worked with pytest, interesting issue. I'd like to have a look as well. Maybe we can in the future just solve this by adding the number to the end of the string instead of the front:

```
  preprocessing_02/
  analysis_03/
```

This way we could still preserve a human-readable order of the scripts.

**Comment by tifa365** [comment]
Maybe a short-term solution could be to add an additional underscore `_`

`_02_preprocessing and 01_split_xml`

Unfortunately I had to rename the whole processing folder, too (I won't add this to my branch yet), to make things work.

`_02_preprocessing`

**Comment by erwee6** [comment]
I would prefer a leading letter to an underscore. „s“ for step or script -> s01_split_xml

**Comment by pmayd** [comment]
No sorry you should not import scripts but functions! When you import the script as is that is not what we should do. You should refactor the current script into functions that live inside a core folder like helper and import it in the script. The same function can then be important in the tests...

**Comment by erwee6** [comment]
So we will never import from the 01 etc. folders?

**Comment by pmayd** [comment]
Exactly, at least not in my understanding and coding, these scripts are main scripts basically using functions that do the actual work and the main scripts are basically simple wrappers as long as we don't use something like classes and OOP, so the main scripts set the start variables like which folders or which input but the rest is all in the end put into functions if that is possible.
